src/libraries/System.Net.Quic/src/System/Net/Quic/Implementations/MsQuic/MsQuicConnection.cs

```diff
@@ -0,0 +1,430 @@
+        {
+            SingleReader = true,
+            SingleWriter = true
+        });
+
+        // constructor for inbound connections
+        public MsQuicConnection(IPEndPoint remoteEndPoint, MsQuicApi api, IntPtr nativeObjPtr)
+        {
+            if (NetEventSource.IsEnabled) NetEventSource.Enter(this);
+            _remoteEndPoint = remoteEndPoint;
+            _api = api;
+            _nativeObjPtr = nativeObjPtr;
+
+            SetCallbackHandler();
+            SetIdleTimeout(TimeSpan.FromSeconds(120));
+            if (NetEventSource.IsEnabled) NetEventSource.Exit(this);
+        }
+
+        // constructor for outbound connections
+        public MsQuicConnection(IPEndPoint remoteEndPoint, MsQuicApi api, IntPtr nativeObjPtr, SslClientAuthenticationOptions sslClientAuthenticationOptions)
+        {
+            if (NetEventSource.IsEnabled) NetEventSource.Enter(this);
+
+            _remoteEndPoint = remoteEndPoint;
+            _nativeObjPtr = nativeObjPtr;
+            _api = api;
+            SetCallbackHandler();
+
+            if (NetEventSource.IsEnabled) NetEventSource.Exit(this);
+        }
+
+        internal override IPEndPoint LocalEndPoint => throw new NotImplementedException();
+
+        internal override IPEndPoint RemoteEndPoint => new IPEndPoint(_remoteEndPoint.Address, _remoteEndPoint.Port);
+
+        internal override SslApplicationProtocol NegotiatedApplicationProtocol => throw new NotImplementedException();
+
+        internal override bool Connected => throw new NotImplementedException();
+
+        internal uint HandleEvent(ref ConnectionEvent connectionEvent)
+        {
+            uint status = MsQuicConstants.Success;
+            try
+            {
+                switch (connectionEvent.Type)
+                {
+                    case QUIC_CONNECTION_EVENT.CONNECTED:
+                        {
+                            status = HandleEventConnected(
+                                connectionEvent);
+                        }
+                        break;
+
+                    case QUIC_CONNECTION_EVENT.SHUTDOWN_BEGIN:
+                        {
+                            status = HandleEventShutdownBegin(
+                                connectionEvent);
+                        }
+                        break;
+
+                    case QUIC_CONNECTION_EVENT.SHUTDOWN_BEGIN_PEER:
+                        {
+                            status = HandleEventShutdownBeginPeer(
+                                connectionEvent);
+                        }
+                        break;
+
+                    case QUIC_CONNECTION_EVENT.SHUTDOWN_COMPLETE:
+                        {
+                            status = HandleEventShutdownComplete(
+                                connectionEvent);
+                        }
+                        break;
+
+                    case QUIC_CONNECTION_EVENT.NEW_STREAM:
```

Comment:
This event is call PEER_STREAM_STARTED now. I'd go and double check all your other names as well.